arcgis-ios-sdk-samples/Geometry/Buffer list/BufferListViewController.swift

```diff
@@ -0,0 +1,184 @@
+        // Clear existing buffers graphics before drawing.
+        bufferGraphicsOverlay.graphics.removeAllObjects()
+        // Create the buffers.
+        // Notice: the radius distances has the same unit of the map's spatial reference's unit.
+        if let bufferPolygon = AGSGeometryEngine.bufferGeometries(tappedPointsAndRadius.map { $0.point }, distances: tappedPointsAndRadius.map { NSNumber(value: $0.radius) }, unionResults: isUnionSwitch.isOn) {
+            let graphics = bufferPolygon.map { AGSGraphic(geometry: $0, symbol: nil) }
+            bufferGraphicsOverlay.graphics.addObjects(from: graphics)
+            setStatus(message: "Buffers created.")
+        }
+    }
+    
+    @IBAction func clearButtonTapped(_ sender: UIBarButtonItem) {
+        bufferGraphicsOverlay.graphics.removeAllObjects()
+        tapLocationsGraphicsOverlay.graphics.removeAllObjects()
+        tappedPointsAndRadius.removeAll()
+        setStatus(message: "Buffers removed. Tap on the map to add buffers.")
+    }
+    
+    @IBAction func distanceSliderValueChanged(_ sender: UISlider) {
+        // Update the buffer radius with the slider value.
+        bufferRadius.value = Double(sender.value)
+    }
+    
+    // MARK: UI
+    
+    func setStatus(message: String) {
+        statusLabel.text = message
+    }
+    
+    // MARK: UIViewController
+     
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["BufferListViewController"]
+        // Load the map image layer.
+        mapImageLayer.load { [weak self] (error) in
+            if let error = error {
+                self?.presentAlert(error: error)
+                self?.setStatus(message: "Fail to load basemap.")
+            } else {
+                self?.setStatus(message: "Tap on the map to add buffers.")
+            }
+        }
+    }
+}
+
+// MARK: - AGSGeoViewTouchDelegate
+
+extension BufferListViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        // Only proceed when the tapped point is within the boundary.
+        guard AGSGeometryEngine.geometry(boundaryPolygon, contains: mapPoint) else {
+            setStatus(message: "Tap within the boundary to add buffer.")
+            return
+        }
+        // The spatial reference in this sample use US feet as unit.
+        let radius = bufferRadius.converted(to: .feet).value
+        // Ensure that the buffer radius is a positive value.
+        guard radius > 0 else { return }
+        // Create and add graphic symbolizing the tap point.
+        let pointGraphic = AGSGraphic(geometry: mapPoint, symbol: nil)
+        tapLocationsGraphicsOverlay.graphics.add(pointGraphic)
+        // Keep track of tapped points and their radius.
```

Comment:
```suggestion
        // Keep track of tapped points and their radii.
```

Reply:
Thanks! This is helpful. I don't know the plural form of radius before... 😅